Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean

```diff
@@ -0,0 +1,354 @@
+  rw [mul_comm]
+  exact le_self_mul
+#align le_mul_self le_mul_self
+
+@[to_additive]
+theorem self_le_mul_right (a b : α) : a ≤ a * b :=
+  le_self_mul
+#align self_le_mul_right self_le_mul_right
+
+@[to_additive]
+theorem self_le_mul_left (a b : α) : a ≤ b * a :=
+  le_mul_self
+#align self_le_mul_left self_le_mul_left
+
+@[to_additive]
+theorem le_of_mul_le_left : a * b ≤ c → a ≤ c :=
+  le_self_mul.trans
+#align le_of_mul_le_left le_of_mul_le_left
+
+@[to_additive]
+theorem le_of_mul_le_right : a * b ≤ c → b ≤ c :=
+  le_mul_self.trans
+#align le_of_mul_le_right le_of_mul_le_right
+
+@[to_additive]
+theorem le_iff_exists_mul : a ≤ b ↔ ∃ c, b = a * c :=
+  ⟨exists_mul_of_le, by
+    rintro ⟨c, rfl⟩
+    exact le_self_mul⟩
+#align le_iff_exists_mul le_iff_exists_mul
+
+@[to_additive]
+theorem le_iff_exists_mul' : a ≤ b ↔ ∃ c, b = c * a := by
+  simp only [mul_comm _ a, le_iff_exists_mul]
+  rfl
+#align le_iff_exists_mul' le_iff_exists_mul'
+
+@[simp, to_additive zero_le]
+theorem one_le (a : α) : 1 ≤ a :=
+  le_iff_exists_mul.mpr ⟨a, (one_mul _).symm⟩
+#align one_le one_le
+
+@[to_additive]
+theorem bot_eq_one : (⊥ : α) = 1 :=
+  le_antisymm bot_le (one_le ⊥)
+#align bot_eq_one bot_eq_one
+
+@[simp, to_additive]
+theorem mul_eq_one_iff : a * b = 1 ↔ a = 1 ∧ b = 1 :=
+  mul_eq_one_iff' (one_le _) (one_le _)
+#align mul_eq_one_iff mul_eq_one_iff
+
+@[simp, to_additive]
+theorem le_one_iff_eq_one : a ≤ 1 ↔ a = 1 :=
+  (one_le a).le_iff_eq
+#align le_one_iff_eq_one le_one_iff_eq_one
+
+@[to_additive]
+theorem one_lt_iff_ne_one : 1 < a ↔ a ≠ 1 :=
+  (one_le a).lt_iff_ne.trans ne_comm
+#align one_lt_iff_ne_one one_lt_iff_ne_one
+
+@[to_additive]
+theorem eq_one_or_one_lt : a = 1 ∨ 1 < a :=
+  (one_le a).eq_or_lt.imp_left Eq.symm
+#align eq_one_or_one_lt eq_one_or_one_lt
+
+@[simp, to_additive add_pos_iff]
+theorem one_lt_mul_iff : 1 < a * b ↔ 1 < a ∨ 1 < b := by
+  simp only [one_lt_iff_ne_one, Ne.def, mul_eq_one_iff, not_and_or]
+  rfl -- Porting note: Should this be needed? It wasn't needed in lean3
+#align one_lt_mul_iff one_lt_mul_iff
+
+@[to_additive]
+theorem exists_one_lt_mul_of_lt (h : a < b) : ∃ (c : _)(hc : 1 < c), a * c = b := by
+  obtain ⟨c, hc⟩ := le_iff_exists_mul.1 h.le
+  refine' ⟨c, one_lt_iff_ne_one.2 _, hc.symm⟩
+  rintro rfl
+  -- Porting note: simpa seems to continue to try and solve the goal after it has been solved
+  simpa [hc, lt_irrefl] using h
+#align exists_one_lt_mul_of_lt exists_one_lt_mul_of_lt
+
+@[to_additive]
+theorem le_mul_left (h : a ≤ c) : a ≤ b * c :=
+  calc
+    a = 1 * a := by simp
+    _ ≤ b * c := mul_le_mul' (one_le _) h
+
+#align le_mul_left le_mul_left
+
+@[to_additive]
+theorem le_mul_right (h : a ≤ b) : a ≤ b * c :=
+  calc
+    a = a * 1 := by simp
+    _ ≤ b * c := mul_le_mul' h (one_le _)
+
+#align le_mul_right le_mul_right
+
+@[to_additive]
+theorem lt_iff_exists_mul [CovariantClass α α (· * ·) (· < ·)] : a < b ↔ ∃ c > 1, b = a * c := by
+  -- Porting note: simp_rw was more aggressive in lean4
```

Comment:
In cases like this, could you leave the original proof in place, commented out? This makes it easier for someone to address the problem, without having to dig up other repositories.